src/polytopes/segment.jl

```diff
@@ -138,18 +138,14 @@ function intersectpoint(s1::Segment{Dim,T}, s2::Segment{Dim,T}) where {Dim,T}
   x1, x2 = vertices(s1)
   y1, y2 = vertices(s2)
 
-  # solve Ax = b
-  A = (y2 - y1) - (x2 - x1)
-  b = (x1 - y1)
-  s = zero(T)
-  for i in 1:length(A)
-    if !iszero(A[i])
-      s = b[i] / A[i]
-      break
-    end
-  end
-
-  x1 + s*(x2 - x1)
+  v1 = x1 - x2
+  v2 = y1 - y2
+  v12 = x1 - y1
+
+  # find scalar along first segment
+  s = (v12[1] * v2[2] - v12[2] * v2[1]) / (v1[1] * v2[2] - v1[2] * v2[1])
```

Comment:
Please break this line into two lines with numerator and denominator. After that and the comment with links to wikipedia we are ready to merge 🎉